addons/sourcemod/scripting/shavit-misc.sp

```diff
@@ -2096,14 +2132,20 @@ public Action Command_Specs(int client, int args)
 	return Plugin_Handled;
 }
 
-void ClearClientEventsFrame(int serial)
+public MRESReturn DHooks_OnTeleport(int pThis, DHookParam hParams)
 {
-	int client = GetClientFromSerial(serial);
-
-	if (client > 0 && gB_Eventqueuefix)
+	if(!IsValidEntity(pThis) || !IsClientInGame(pThis) || IsFakeClient(pThis))
 	{
-		ClearClientEvents(client);
+		return MRES_Ignored;
 	}
+	
+	if(!hParams.IsNull(1))
+	{
+		gB_WasInStartZoneBeforeTeleport[pThis] = Shavit_InsideZone(pThis, Zone_Start, -1);
```

Comment:
it's possible to stand inside a bonus start zone and then restart to the main track which will cause the event clearing to not happen (and no targetname reset too on maps if enabled). might not introduce too many problems. not sure what the best way to deal with that would be but `Shavit_OnRestart` can be used I guess if you want to try tracking teleporting through restart.

Also, using `!sp` to have a spawnpoint outside of the startzone would also end up with the events & targetname not being reset/set/cleared/etc.

Reply:
Not sure how the ``!sp`` outside the start zone could be abused tho? Any examples?

Reply:
Not necessarily sure it could be abused unless the map fails to set targetname at the start. The bonus one too of course might be hard to get an example of it breaking specific maps because it depends on bonus placement & potential boosters that overlap or triggers with delays you could reach the bonus start before it goes off.

Reply:
Nah, I get that problem about bonus and main starts, would play around with it. But didn't get the ``!sp`` problem, isn't it like a cp anyway, and if the map don't have any reset triggers on start simply doing ``!r`` would still cause issues if no reset is done.

Reply:
>isn't it like a cp anyway

it's just `sm_r -> Shavit_RestartTimer(client) (Shavit_StopTimer()) -> shavit-zones::Shavit_OnRestart() -> TeleportEntity() [-> Shavit_StartTimer() when not !sp]`.

>if the map don't have any reset triggers on start simply doing `!r` would still cause issues if no reset is done.

yeah, this probably only ends up being a problem on the like 1 or 2 maps that need the targetname reset which wouldn't run since the shavit_misc_forcetargetnamereset code only happens in the first-tick-since-teleport section

Reply:
But when they would enter the start zone the reset would fire again after that anyway (if cvar is enabled that is), as the code is always run when you just stand on the ground in a start zone.

Reply:
Doesn't the targetname reset only happen within the teleported tick +0 / +1 in this?
```
if((gI_LatestTeleportTick[client] <= curr_tick <= gI_LatestTeleportTick[client] + 1) && 
		!gB_WasInStartZoneBeforeTeleport[client] && curr_tick != tick_served[client])
```

Reply:
Yeah, now I see where this is going, you are right, I'll check this too then.

Reply:
Removed the start zone tracking code to prevent abusing the events from bonuses and vise versa. For the !sp problem, the code that resets targetname was moved back to where it was (and now will be fired on every OnStart() forward call).